### ISSUE TITLE
Prevent re-downloading of previously finished downloads with ParallelDownloader

### ISSUE THREAD

**Issue by azihassan** [azihassan/youtube-d#23]
When attempting to redownload a previously downloaded video, RegularDownloader won't overwrite the existing file. After reading its size in bytes, it will attempt to download starting from that size (with the Range HTTP header) thus resulting in a noop of sorts.

ParallelDownloader only does this with partial files. However, once a download is finished, the partial files are merged into a final video file and then deleted. Redownloading the same video with ParallelDownloader causes it to start over again because it only checks for the presence of partial files.

To fix this, I'm thinking about adding an `hasFinished` method to the `Downloader` interface to make sure that the downloaders perform this verification before doing anything else.
